### PR TITLE
examples: Support DWARF5 filename decoding

### PR DESCRIPTION
**Pull request by cyyself**
DWARF5 uses 0-indexed file_entry and directory_entry.  We should not skip index 0 when decoding the filename and the directory for DWARF5.

### CODE REVIEW

**Comment by sevaa** [comment]
Makes sense.

File: examples/dwarf_lineprogram_filenames.py [hunk 1]
```diff
@@ -77,16 +79,22 @@ def lpe_filename(line_program, file_index):
     lp_header = line_program.header
     file_entries = lp_header["file_entry"]
 
-    # File and directory indices are 1-indexed.
-    file_entry = file_entries[file_index - 1]
+    # File and directory indices are 1-indexed in DWARF version < 5,
+    # 0-indexed in DWARF5.
+    file_index -= 1 if lp_header.version < 5 else 0
```
Comment (eliben) [inline examples/dwarf_lineprogram_filenames.py:84]: Please rewrite this with simpler logic:

```
if ...version...:
  file_index -= 1
```

File: examples/dwarf_lineprogram_filenames.py [hunk 2]
```diff
@@ -77,16 +79,22 @@ def lpe_filename(line_program, file_index):
...
 
     # A dir_index of 0 indicates that no absolute directory was recorded during
-    # compilation; return just the basename.
-    if dir_index == 0:
+    # compilation in DWARF version < 5; return just the basename.
+    if dir_index == 0 and lp_header.version < 5:
         return file_entry.name.decode()
 
-    directory = lp_header["include_directory"][dir_index - 1]
+    dir_index -= 1 if lp_header.version < 5 else 0
```
Comment (eliben) [inline examples/dwarf_lineprogram_filenames.py:96]: Same here

**Review by eliben — CHANGES_REQUESTED** [review]
Thank you

Just a minor code style request

**Comment by cyyself** [comment]
> Thank you
> 
> Just a minor code style request

Resolved now. Thanks!

**Review by eliben — APPROVED** [review]

**Pull request merged**